modules/core/js/src/main/scala/com/azavea/stac4s/ItemProperties.scala

```diff
@@ -0,0 +1,119 @@
+package com.azavea.stac4s
```

Comment:
This code duplication is an _extreme_ drag -- I don't know how to get the collision of `java.time.Instant` classpaths to cooperate in a way that can save us this effort. I needed to do the JS implementation here because `ItemExtension` is in `shared/` and implementing these changes was easier than documenting the ways that they're different.

Reply:
@jisantuc hmm, I put `ItemProperties` and `ItemDatetime` into the `shared` and all tests just pass. Why is it required to duplicate these classes?

Reply:
Oh, I thought the classpath collision with `java.time.Instant` caused us problems before. Great! I will move that then.

Reply:
What a delightful thing to have been wrong about 😎

Thanks for testing it out and not just trusting me that we had to copy so much stuff